frontend/src/core/editor/selectors.ts

```diff
@@ -6,22 +6,21 @@ import { fluent } from 'core/utils';
 import * as history from 'modules/history';
 import { NAME } from '.';
 
-import type { EntityTranslation, Entity } from 'core/api';
-import type { HistoryTranslation, HistoryState } from 'modules/history';
-import type { EditorState } from '.';
+import type { RootState } from '../../store';
 
-const editorSelector = (state): EditorState => state[NAME];
-const historySelector = (state): HistoryState => state[history.NAME];
+const editorSelector = (state: RootState) => state[NAME];
+const historySelector = (state: RootState) => state[history.NAME];
 
 export function _existingTranslation(
-    editor: EditorState,
-    activeTranslation: EntityTranslation,
-    history: HistoryState,
-    entity: Entity,
-): (EntityTranslation | HistoryTranslation) | null | undefined {
+    editor: ReturnType<typeof editorSelector>,
+    activeTranslation: ReturnType<
+        typeof plural.selectors.getTranslationForSelectedEntity
+    >,
+    history: ReturnType<typeof historySelector>,
+    entity: ReturnType<typeof entities.selectors.getSelectedEntity>,
+) {
```

Comment:
Is the return type intentionally dropped here?

Reply:
Yes, TS is good about deducing it. And leaving out `null` and `undefined` should help when we transition to `strictNull`